src/views/Institutions/Institution.jsx

```diff
@@ -57,36 +62,79 @@ const Institute = (props) => {
     }
 
     // need to remove id and show from the payload
-    let {id, show, created_at, updated_at, created_by_frontend, updated_by_frontend, ...dataToSave} = data;
-    if (typeof data.logo === 'object') {
-      dataToSave['logo'] = data.logo?.id;
+    let {
+      id,
+      show,
+      mou_list,
+      created_at,
+      updated_at,
+      created_by_frontend,
+      updated_by_frontend,
+      ...dataToSave
+    } = data;
+    if (typeof data.logo === "object") {
+      dataToSave["logo"] = data.logo?.id;
+    }
+
+    if (mou_list && mou_list.length) {
+      dataToSave['mou_list'] = []
+      await Promise.all(
+        mou_list.map(async (mouData) => {
+          try  {
+            const {
+              data: {
+                data: {
+                    upload: { id: uploadedMouId }
+                  }
+              }
+            } = await uploadFile(mouData.mou)
+
+            dataToSave['mou_list'].push({
+              ...mouData,
+              mou: uploadedMouId
```

Comment:
let's make reduce the code size
```suggestion
            const response = await uploadFile(mouData.mou)
            dataToSave['mou_list'].push({
              ...mouData,
              mou: response.data.data.upload.id
```